dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -206,15 +209,17 @@ const TableWithFavorite = () => {
                 <Tbody>
                   {selectedArray.length > 0 ? (
                     selectedArray.map((repo, rowIndex) => (
-                      <Tr key={rowIndex}>
+                      <Tr key={repo?.resource_id}>
                         <Td
                           dataLabel={columnNames.name}
-                          onClick={() => navigate(`${TOC}/${repo.resource_id}`)}
+                          onClick={() =>
+                            navigate(`${TOC}/${repo?.resource_id}`)
+                          }
                         >
-                          {repo.name}
+                          {repo?.name}
                         </Td>
                         <Td dataLabel={columnNames.uploadedDate}>
-                          {repo.metadata[DATASET_UPLOADED]}
+                          {repo?.metadata[DATASET_UPLOADED]}
```

Comment:
I don't think we want to see `undefined` from any of these `repo` references, so I don't think we should be using the chaining operator here.

Reply:
Chaining is necessary here. As the entries are stored in  _sparse_ array, when we try to render the page by reading properties of undefined it throws an error.

Reply:
I'm confused:  if the code would have thrown an error without the chaining reference, then what does it display to the user if you use the chaining reference?  (Yes, the entries are stored in a sparse array, but the part that we're displaying should be fully populated...right?)

It seems like, if the code would throw an error, then something is broken (i.e., we don't yet have the data that we're supposed to be displaying, so we shouldn't be trying to display it).  If we have our ducks in a row, then it shouldn't matter which operator we use, and so we should use the normal references, so that, when things aren't working, we get an error instead of undefined behavior.